runhouse/cli_utils.py

```diff
@@ -0,0 +1,475 @@
+        for gpu_type in gpu_types:
+            gpu_types_str = gpu_types_str + f", {gpu_type}"
+        cpus_gpus_info_str = (
+            cpus_gpus_info_str + f" | GPUs: {num_of_gpus} (Type(s): {gpu_types_str})"
+        )
+    console.print(cpus_gpus_info_str)
+
+
+def print_status(status_data: dict, current_cluster) -> None:
+    from runhouse.main import console
+
+    """Prints the status of the cluster to the console"""
+    cluster_config = status_data.get("cluster_config")
+    env_servlet_processes = status_data.get("env_servlet_processes")
+
+    cluster_name = cluster_config.get("name", None)
+
+    if cluster_name:
+        api_server_url = cluster_config.get(
+            "api_server_url", rh.configs.get("api_server_url")
+        )
+        api_server_url = api_server_url.replace(
+            "api", "www"
+        )  # convert the api link to the ui link.
+        cluster_link_in_den_ui = get_resource_link_in_den_ui(
+            cluster_name=cluster_name, api_server_url=api_server_url
+        )
+        cluster_name_hyperlink = rich.markdown.Text(
+            cluster_name, style=f"link {cluster_link_in_den_ui} white"
+        )
+        console.print(cluster_name_hyperlink)
+
+    has_cuda: bool = cluster_config.get("has_cuda")
+
+    # print headline
+    daemon_headline_txt = (
+        "\N{smiling face with horns} Runhouse Daemon is running \N{Runner}"
+    )
+    console.print(daemon_headline_txt, style="bold royal_blue1")
+
+    console.print(f"Runhouse v{status_data.get('runhouse_version')}")
+    print_cloud_properties(cluster_config)
+    console.print(f"server pid: {status_data.get('server_pid')}")
+
+    # Print relevant info from cluster config.
+    print_cluster_config(cluster_config)
+
+    # print general cpu and gpu utilization
+    cluster_gpu_utilization: float = status_data.get("server_gpu_utilization")
+
+    # cluster_gpu_utilization can be none, if the cluster was not using its GPU at the moment cluster.status() was invoked.
+    if cluster_gpu_utilization is None and has_cuda:
+        cluster_gpu_utilization: float = 0.0
+
+    cluster_cpu_utilization: float = status_data.get("server_cpu_utilization")
+
+    server_util_info = (
+        f"CPU Utilization: {round(cluster_cpu_utilization, 2)}% | GPU Utilization: {round(cluster_gpu_utilization, 2)}%"
+        if has_cuda
+        else f"CPU Utilization: {round(cluster_cpu_utilization, 2)}%"
+    )
+    console.print(server_util_info)
+
+    # print the environments in the cluster, and the resources associated with each environment.
+    print_envs_info(env_servlet_processes, current_cluster)
+
+
+def print_bring_cluster_up_msg(
+    cluster_name: str, msg_prefix="Can't execute the command"
+):
+    from runhouse.main import console
+
+    console.print(
+        f"{msg_prefix} because [reset]{cluster_name} is terminated. To bring it up, run [bold italic green]runhouse cluster up {cluster_name}[/bold italic green]."
```

Comment:
i don't think we always no for sure that the cluster status is terminated when we print this message? 
maybe better to change to "not up"